packages/netlify-cms-core/src/components/MediaLibrary/MediaLibrary.js

```diff
@@ -44,9 +44,13 @@ class MediaLibrary extends React.Component {
     if (isOpening) {
       this.setState({ selectedFile: {}, query: '' });
     }
+  }
+
+  componentDidUpdate(prevProps) {
+    const isOpening = !prevProps.isVisible && this.props.isVisible;
 
-    if (isOpening && (this.props.privateUpload !== nextProps.privateUpload)) {
-      this.props.loadMedia({ privateUpload: nextProps.privateUpload });
+    if (isOpening && (prevProps.privateUpload !== this.props.privateUpload)) {
+      this.props.loadMedia({ privateUpload: this.props.privateUpload });
```

Comment:
[Side effects on props change should be in `componentDidUpdate`](https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html#side-effects-on-props-change)